fairscale/experimental/nn/distributed_pipeline/trace.py

```diff
@@ -0,0 +1,101 @@
+import inspect
+import operator
+from typing import Dict, Optional, cast
+
+from torch.distributed.nn import RemoteModule
+import torch.fx
+import torch.nn as nn
+
+from . import PipelineModulesGraph
+
+
+class RemoteModuleTracer(torch.fx.Tracer):
+    def is_leaf_module(self, m: torch.nn.Module, module_qualified_name: str) -> bool:
+        if isinstance(m, RemoteModule):
+            return True
+        return False
+
+
+class GraphCreator:
+    def __init__(self, tracer: RemoteModuleTracer) -> None:
+        self.tracer = tracer
+
+    def get_module(self, node: torch.fx.Node) -> Optional[nn.Module]:
+        """Given a call_module node, returns the module corresponding to this module call"""
+        if node.op != "call_module":
+            return None
+        module = self.tracer.root
+        for t in cast(str, node.target).split("."):
+            module = getattr(module, t)
```

Comment:
Do you just want to look at the last element in the array that split returns instead of enumerating through them?

Reply:
Say if original string is 'a.b.c.d', we need to return self.tracer.root.a.b.c.d. So we iterator over the array ['a', 'b', 'c', 'd'] to start from self.tracer.root and get to the module with name 'd'.